File: 2022/04/t.py
cal = open('input.txt', 'r', encoding='utf-8').read().strip().split('\n')

soma = 0
soma2 = 0
for txt in cal:
    a = txt.split(',')
    b = a[0].split('-')
    c = a[1].split('-')

    if ((int(b[0]) >= int(c[0])) and (int(b[1]) <= int(c[1]))) or \
            ((int(c[0]) >= int(b[0])) and (int(c[1]) <= int(b[1]))):
        soma += 1

    if not ((int(b[1]) < int(c[0])) or (int(c[1]) < int(b[0]))):
        soma2 += 1

print('1', soma)
print('2', soma2)

File: 2023/01/t.java
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

public class t {
    public static int Soma(String dados) {
        int sum = 0;
        for (String block : dados.split("\r?\n")) {
            String onlyNum = block.replaceAll("[a-z]", "");
            sum += ((onlyNum.charAt(0) - '0') * 10) + (onlyNum.charAt(onlyNum.length() - 1) - '0');
        }
        return sum;
    }

    public static int Ajusta(String dados) {
        String[] nums = {"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        for (int i = 1; i <= 9; i++) {
            String numStr = nums[i];
            dados = dados.replaceAll(numStr, numStr.charAt(0) +  Integer.toString(i) + numStr.charAt(numStr.length() - 1));
        }
        return Soma(dados);
    }

    public static void main(String[] args) throws IOException {

        String content = Files.readString(Paths.get("input.txt"), StandardCharsets.UTF_8).strip();

        System.out.printf("Round 1: %s%n", Soma(content)); // 55386
        System.out.printf("Round 2: %s%n", Ajusta(content)); // 54824
    }
}
